src/eval/execution.rs

```diff
@@ -0,0 +1,198 @@
+            "Could not find the nix value: {:?}",
+            namespace_obj.to_rust_string_lossy(scope)
+        )
+    };
+    let nix_value: v8::Local<v8::Function> =
+        nix_value.try_into().expect("Nix value must be a function.");
+
+    let nixrt_attr = v8::String::new(scope, "__nixrt").unwrap();
+    let nixrt: v8::Local<v8::Value> = namespace_obj.get(scope, nixrt_attr.into()).unwrap();
+
+    let eval_ctx = create_eval_ctx(
+        scope,
+        &nixrt,
+        &current_dir().map_err(|err| {
+            format!("Failed to determine the current working directory. Error: {err}")
+        })?,
+    )?;
+
+    let nix_value = call_js_function(scope, &nix_value, &[eval_ctx.into()])?;
+
+    let to_strict_fn: v8::Local<v8::Function> = try_get_js_object_key(scope, &nixrt, "toStrict")?
+        .expect("Could not find the function `toStrict` in `nixrt`.")
+        .try_into()
+        .expect("`n.toStrict` is not a function.");
+    let strict_nix_value = call_js_function(scope, &to_strict_fn, &[nix_value])?;
+
+    js_value_to_nix(scope, &nixrt, &strict_nix_value)
+}
+
+fn call_js_function<'s>(
+    scope: &mut v8::ContextScope<'_, v8::HandleScope<'s>>,
+    js_function: &v8::Local<v8::Function>,
+    args: &[v8::Local<v8::Value>],
+) -> Result<v8::Local<'s, v8::Value>, String> {
+    let scope = &mut v8::TryCatch::new(scope);
+    let recv = v8::undefined(scope).into();
+    let Some(strict_nix_value) = js_function.call(scope, recv, args) else {
+        // TODO: Again, the stack trace needs to be source-mapped. See TODO above.
+        let err_msg = scope
+            .stack_trace()
+            .map_or("Unknown evaluation error.".to_owned(), |stack| {
+                stack.to_rust_string_lossy(scope)
+            });
+        return Err(err_msg);
+    };
+    Ok(strict_nix_value)
+}
+
+fn create_eval_ctx<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    nixrt: &v8::Local<v8::Value>,
+    script_path: &Path,
+) -> Result<v8::Local<'s, v8::Object>, String> {
+    let eval_ctx_type = get_nixrt_type(scope, nixrt, "EvalCtx")?;
+    let eval_ctx_constructor: v8::Local<v8::Function> = eval_ctx_type
+        .try_into()
+        .expect("Could not get the constructor of the evaluation context class.");
+
+    let real_path = script_path
+        .canonicalize()
+        .map_err(|err| format!("Failed to resolve the script path. Error: {err}."))?;
+    let script_dir = real_path
+        .parent()
+        .ok_or_else(|| format!("Failed to determine the directory of path {real_path:?}."))?;
+    let script_dir_str = real_path
+        .to_str()
+        .ok_or_else(|| format!("Failed to converft the path {script_dir:?} to a string."))?;
+    let js_script_dir_path =
+        v8::String::new(scope, script_dir_str).expect("Unexpected internal error.");
+
+    Ok(eval_ctx_constructor
+        .new_instance(scope, &[js_script_dir_path.into()])
+        .expect("Could not construct the global evaluation context."))
+}
+fn new_script_origin<'s>(
```

Comment:
Funny, I'd have thought `cargo fmt` would add a newline here.
```suggestion

fn new_script_origin<'s>(
```